extensions/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java

```diff
@@ -232,13 +209,62 @@ public void register(RegistrationContext registrationContext) {
         restClientRecorder.setSslEnabled(sslNativeConfig.isEnabled());
     }
 
-    private ScopeInfo computeDefaultScope(Config config, Map.Entry<DotName, ClassInfo> entry) {
-        DotName restClientName = entry.getKey();
+    private void findInterfaces(IndexView index, Map<DotName, ClassInfo> interfaces, Set<Type> returnTypes,
+            DotName annotationToFind) {
+        for (AnnotationInstance annotation : index.getAnnotations(annotationToFind)) {
+            AnnotationTarget target = annotation.target();
+            ClassInfo theInfo;
+            if (target.kind() == AnnotationTarget.Kind.CLASS) {
+                theInfo = target.asClass();
+            } else if (target.kind() == AnnotationTarget.Kind.METHOD) {
+                theInfo = target.asMethod().declaringClass();
+            } else {
+                continue;
+            }
+
+            if (!isRestClientInterface(index, theInfo)) {
+                continue;
+            }
+
+            interfaces.put(theInfo.name(), theInfo);
+
+            // Find Return types
+            processInterfaceReturnTypes(theInfo, returnTypes);
+            for (Type interfaceType : theInfo.interfaceTypes()) {
+                ClassInfo interfaceClassInfo = index.getClassByName(interfaceType.name());
+                if (interfaceClassInfo != null) {
+                    processInterfaceReturnTypes(interfaceClassInfo, returnTypes);
+                }
+            }
+        }
+    }
+
+    private void processInterfaceReturnTypes(ClassInfo classInfo, Set<Type> returnTypes) {
+        for (MethodInfo method : classInfo.methods()) {
+            Type type = method.returnType();
+            if (!type.name().toString().contains("java.lang")) {
```

Comment:
nitpick: `startsWith` would be safer  than `contains`, you never know what kind of insane person could use a package that contains `java.lang` somewhere in the middle

Reply:
Fixed